### PR TITLE
Fixing Trainer with u8 instead of chars.

### PR DESCRIPTION
**Pull request by Narsil**
Now check both optimized and unoptimized encodings schemes for Unigram.

### CODE REVIEW

File: tokenizers/src/models/unigram/lattice.rs [hunk 1]
```diff
@@ -220,11 +226,14 @@ impl<'a> Lattice<'a> {
     }
 
     pub fn piece(&self, node: &Node) -> String {
-        self.sentence
-            .chars()
-            .skip(node.pos)
-            .take(node.length)
-            .collect()
+        String::from_utf8(
```
Comment (n1t0) [inline tokenizers/src/models/unigram/lattice.rs:229]: Can't we do `self.sentece[node.pos..node.pos+length].to_owned()`?

File: tokenizers/src/models/unigram/model.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ pub struct Unigram {
     pub(super) eos_id: usize,
 
     fuse_unk: bool,
+    is_optimized: bool,
```
Comment (n1t0) [inline tokenizers/src/models/unigram/model.rs:27]: This is going to be serialized, maybe you want to `#[serde(skip)]` it

File: tokenizers/tokenizer.json [hunk 3]
```diff
@@ -0,0 +1,194 @@
+{
```
Comment (n1t0) [inline tokenizers/tokenizer.json:1]: Haha, nope!

**Review by n1t0 — APPROVED** [review]
Just a couple things to verify (see comments below), but otherwise LGTM.

File: tokenizers/tokenizer.json (hunk 3 above)
Comment (Narsil) [reply tokenizers/tokenizer.json:1]: Ahah

**Pull request merged**